enclave-tls/src/tls_wrappers/openssl/use_privkey.c

```diff
@@ -18,8 +18,16 @@ tls_wrapper_err_t openssl_tls_use_privkey(tls_wrapper_ctx_t *ctx, void *privkey_
 
 	openssl_ctx_t *ssl_ctx = (openssl_ctx_t *)ctx->tls_private;
 
-	int ret = SSL_CTX_use_PrivateKey_ASN1(EVP_PKEY_RSA, ssl_ctx->sctx, privkey_buf,
-					      (long)privkey_len);
+	int ret;
+
+	if (algo == ENCLAVE_TLS_CERT_ALGO_ECC_256_SHA256) {
+		ret = SSL_CTX_use_PrivateKey_ASN1(EVP_PKEY_EC, ssl_ctx->sctx, privkey_buf,
+						      (long)privkey_len);
+	} else if (algo == ENCLAVE_TLS_CERT_ALGO_RSA_3072_SHA256) {
+		ret = SSL_CTX_use_PrivateKey_ASN1(EVP_PKEY_RSA, ssl_ctx->sctx, privkey_buf,
+						      (long)privkey_len);
+	}
+
```

Comment:
Please define a new variable for first argument, and reuse the callling function out of `if` scope